src/generator/AutoRest.Go/TransformerGo.cs

```diff
@@ -0,0 +1,258 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// 
+
+using AutoRest.Core;
+using AutoRest.Core.Logging;
+using AutoRest.Core.Model;
+using AutoRest.Core.Utilities;
+using AutoRest.Extensions;
+using AutoRest.Go.Model;
+using AutoRest.Go.Properties;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+
+namespace AutoRest.Go
+{
+    public class TransformerGo : CodeModelTransformer<CodeModelGo>
+    {
+        private readonly Dictionary<IModelType, IModelType> _normalizedTypes;
+
+        public TransformerGo()
+        {
+            _normalizedTypes = new Dictionary<IModelType, IModelType>();
+        }
+
+        public override CodeModelGo TransformCodeModel(CodeModel cm)
+        {
+            var cmg = cm as CodeModelGo;
+
+            SwaggerExtensions.ProcessGlobalParameters(cmg);
+            // Add the current package name as a reserved keyword
+            CodeNamerGo.Instance.ReserveNamespace(cm.Namespace);
+            FixStutteringTypeNames(cmg);
+            TransformEnumTypes(cmg);
+            TransformMethods(cmg);
+            TransformModelTypes(cmg);
+
+            return cmg;
+        }
+
+        private void TransformEnumTypes(CodeModelGo cmg)
+        {
+            // fix up any enum types that are missing a name.
+            // NOTE: this must be done before the next code block
+            foreach (var mt in cmg.ModelTypes)
```

Comment:
I think the block here on line 48 and the block on 66 should have matching style, either in Linq or foreach blocks.